common/autoware_universe_utils/include/autoware/universe_utils/geometry/geometry.hpp

```diff
@@ -583,6 +584,148 @@ std::optional<geometry_msgs::msg::Point> intersect(
  */
 bool intersects_convex(const Polygon2d & convex_polygon1, const Polygon2d & convex_polygon2);
 
+// Alternatives for Boost.Geometry ----------------------------------------------------------------
```

Comment:
(minor) I think the new code could be added to a separate file as the current one is becoming too big.
This can be done in a future PR.